pydantic/_internal/_std_types_schema.py

```diff
@@ -66,26 +69,47 @@ def to_enum(__input_value: Any, **_kwargs: Any) -> Enum:
         except ValueError:
             raise PydanticCustomError('enum', 'Input is not a valid enum member')
 
-    literal_schema = core_schema.literal_schema(*[m.value for m in enum_type.__members__.values()])
-
-    if issubclass(enum_type, IntEnum):
+    literal_schema = core_schema.literal_schema(
+        *[m.value for m in enum_type.__members__.values()],
+        ref=f'{getattr(enum_type, "__module__", None)}.{enum_type.__name__}',
+    )
+    json_schema_misc = JsonSchemaMisc(
+        core_schema_override=literal_schema.copy(),
+        title=enum_type.__name__,
+        description=inspect.cleandoc(enum_type.__doc__ or 'An enumeration.'),
+    )
+    # TODO: Is 'enum_json_schema_misc' the best name? Is there a better way to handle this?
+    if hasattr(enum_type, 'enum_json_schema_misc'):
+        # The enum_json_schema_extra method should be a classmethod with signature:
+        #   def enum_json_schema_extra(cls, default: JsonSchemaMisc) -> JsonSchemaMisc: ...
+        # For convenience, the value generated above is passed to this function so you can use some/all of its defaults
+        json_schema_misc = getattr(enum_type, 'enum_json_schema_misc')(json_schema_misc)
```

Comment:
Also, I think we could do something similar for arbitrary types, but I'm inclined not to unless there's a really good argument for it; just another thing to maintain. (I could easily be convinced to just drop the `enum_json_schema_misc` method thing entirely, but I feel it's worth putting the idea out there since I already added it.)

Reply:
I realized __modify_schema__ (or whatever we rename it to) should work for this.